src/animation/interpolants/SmoothInterpolant.js

```diff
@@ -0,0 +1,103 @@
+/**
+ *
+ * Cubic hermite spline interpolant.
+ *
+ *
+ * @author tschw
+ */
+
+THREE.SmoothInterpolant = function( times, values, stride, result ) {
+
+	THREE.Interpolant.call( this, times, values, stride, result );
+
+};
+
+Object.assign( THREE.SmoothInterpolant.prototype, THREE.Interpolant.prototype, {
+
+	constructor: THREE.SmoothInterpolant,
+
+	DefaultParameters: {
```

Comment:
> > The values you are referring to are Boolean ones, that specify how the spline behaves at the ends when running out of keyframes.

> you can add extra keys at the beginning and the end. 

You can. But once you add keys for a smooth start / end, the sequence won't loop smoothly anymore. Conversely, once you prep a clip for looping, it wont have a smooth start / stop.

> Is there a reference design you are following?

Yes, I am following *your* design: Since the current system defines the loop mode as part of the Action and not the Clip, we must be able to smoothly start / stop and loop, right? Ideally without duplicating all the data.

> I thought you could do all that with just additional keys -- 

You can, once you implement loop points, so there's in fact a part before the looping and after the looping that the artist can specify. Then just using natural splines (the default) for everything will do.

> I guess this is an implementation choice?

Loop points can be implemented in the Mixer or on top of it. I did not make that choice, but left it open.
I provided a low level interface that works for all cases but didn't wire it up. 

> For another reference here is Unity's AnimationCurve, which is analogous:
> http://docs.unity3d.com/ScriptReference/AnimationCurve.html

These are weighted, so we'd need an extra scalar per keyframe and another interpolant for that data.